temporary is written but not read in constructor

mumbel opened the pull request:
Ran `$ find ~/git/ghidra/Ghidra/Processors/ -name "*.slaspec" -exec ~/ghidra_9.1.1_PUBLIC/support/sleigh -x -t {} /tmp/foo 2>/dev/null \; | grep "temporary is written but not read in constructor"
`
And tried to fix results, some fixes are questionable and need review, but definitely found bugs

fixes #1458

GhidorahRex commented on Ghidra/Processors/SuperH4/data/languages/SuperH4.sinc:
```diff
@@ -2374,8 +2374,8 @@ define pcodeop mac_wOp;
 	N_0t = temp - zext($(T_FLAG));
-	$(T_FLAG) = ( 0 < temp );
-	if ( temp >= N_0t ) goto <skip>;
+	$(T_FLAG) = ( 0 s< temp );
```
According to the manual, temp is an unsigned long, so these two checks should be unsigned.

mumbel replied:
Ah, okay... There was just a warning there.  so if a change were made, maybe `temp != 0`

GhidorahRex replied:
That would be a solid improvement.

mumbel commented on Ghidra/Processors/Atmel/data/languages/avr32a.slaspec:
```diff
@@ -696,7 +696,7 @@ macro subflags(OP1, OP2, RES) {
         #    (!(OP1[31,1]) && RES[31,1]);
 
-        C = OP1 < OP2;
+        C = OP1 s< OP2;
```
I really regret not writing down my reasoning for each commit... ill revert this one

GhidorahRex replied:
If it makes you feel better, compare the pcode with the answer at https://stackoverflow.com/questions/17749946/how-to-implement-borrow-flag-in-c
This matches how we implement the unsigned borrow in the pcoderef.html documentation (see INT_SBORROW)

mumbel commented on Ghidra/Processors/8051/data/languages/8051_main.sinc:
```diff
@@ -365,9 +365,9 @@ macro push8(val) {
+macro push24(val) { val = val; }
 @else
-macro push16(val) { }
+macro push16(val) { val = val; }
```
Is this an acceptable way to handle these?  I can go through later today and add this logic

GhidorahRex replied:
I think it's acceptable. I'm not a fan of having an empty macro, although I don't believe it actually hurts anything.

mumbel commented:
I addressed your comments, thanks for taking thorough look.  I went ahead and squashed, hopefully that is okay.

I did find a few more empty macros and added those.  `8048.slaspec` and then a few others that you had pointed out in `8051_main.sinc`.

Also made sure there were anymore of the other types of issues I encountered, but didn't find any (chalk it up to bad grep foo or just no more)

GhidorahRex commented:
@mumbel that looks great, thanks!
